source/core/ib_stencils.cc

```diff
@@ -0,0 +1,235 @@
+
+      Point<dim, double> interpolation_point_2(dof_point + vect_ib * 1 / 12.);
+
+      Point<dim, double> interpolation_point_3(dof_point + vect_ib * 1. / 8.);
+
+      interpolation_points.resize(3);
+      interpolation_points[0] = interpolation_point_1;
+      interpolation_points[1] = interpolation_point_2;
+      interpolation_points[2] = interpolation_point_3;
+    }
+  if (order == 4)
+    {
+      point = dof_point;
+      Tensor<1, dim, double> vect_ib =
+        (dof_point - p.position -
+         p.radius * (dof_point - p.position) / (dof_point - p.position).norm());
+
+      Point<dim, double> interpolation_point_1(dof_point + vect_ib * 1. / 32.);
+
+      Point<dim, double> interpolation_point_2(dof_point + vect_ib * 1 / 16.);
+
+      Point<dim, double> interpolation_point_3(dof_point + vect_ib * 3. / 32.);
+
+      Point<dim, double> interpolation_point_4(dof_point + vect_ib * 1. / 8.);
+
+      interpolation_points.resize(4);
+      interpolation_points[0] = interpolation_point_1;
+      interpolation_points[1] = interpolation_point_2;
+      interpolation_points[2] = interpolation_point_3;
+      interpolation_points[3] = interpolation_point_4;
+    }
+  if (order > 4)
+    {
+      // In this case the cell is directly used to find the solution at the IB
+      // position. In this case only one point is needed (the position of the
+      // point on the IB).
+      Tensor<1, dim, double> vect_ib =
+        (dof_point - p.position -
+         p.radius * (dof_point - p.position) / (dof_point - p.position).norm());
+
+      point = dof_point - vect_ib;
+
+      Point<dim, double> interpolation_point_1(dof_point + vect_ib * 1. / 8.);
+
+      interpolation_points.resize(1);
+      interpolation_points[0] = interpolation_point_1;
+    }
+  return {point, interpolation_points};
+}
+
+
+template <int dim>
+double
+IBStencils<dim>::ib_velocity(IBParticle<dim> p,
+                             Point<dim>      dof_point,
+                             unsigned int    component)
+{
+  // Return the value of the IB condition for that specific stencil.
+  double v_ib = 0;
+
+  Tensor<1, 3, double> radius;
+  if (dim == 2)
+    {
+      // have to do that conversion as there is no proper conversion from tensor
+      // of dim 2 to 3.
+      radius[0]          = p.radius * ((dof_point - p.position) /
+                              (dof_point - p.position).norm())[0];
+      radius[1]          = p.radius * ((dof_point - p.position) /
+                              (dof_point - p.position).norm())[1];
+      radius[2]          = 0;
+      Tensor<1, 3> v_rot = cross_product_3d(p.omega, radius);
+      if (component == 0)
+        {
+          // vx in 2D
+          v_ib = v_rot[0] + p.velocity[0];
+        }
+      if (component == 1)
+        {
+          // vy in 2D
+          v_ib = v_rot[1] + p.velocity[1];
+        }
+    }
+  if (dim == 3)
+    {
+      ;
+      radius[0]          = p.radius * ((dof_point - p.position) /
```

Comment:
What is radius here? Is it like a distance vector? Radius to me is a particle property no?

Reply:
it's the vector from the center of the particle to the surface close to the DOF.  should it be reference as radial vector instead

Reply:
Radial_vector (in lower case) is a very good way to name it.

Reply:
applied